File: data/src/main/java/org/oppia/data/backends/gae/api/StoryService.java
package org.oppia.data.backends.gae.api;

import org.oppia.data.backends.gae.model.Story;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface StoryService {

  @GET("story_data_handler/{story_id}")
  Call<Story> getStory(@Path("story_id") String storyId, @Query("user_id") String userId, @Query("user") String user);

}

File: data/src/main/java/org/oppia/android/data/backends/gae/api/StoryService.kt
package org.oppia.android.data.backends.gae.api

import org.oppia.android.data.backends.gae.model.GaeStory
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

/** Service that provides access to story endpoints. */
interface StoryService {

  @GET("story_data_handler/{story_id}")
  fun getStory(
    @Path("story_id") storyId: String,
    @Query("user_id") userId: String?,
    @Query("user") user: String?
  ): Call<GaeStory>
}
